### DISCUSSION TITLE
RS485 communication via I2C and ethernet

### DISCUSSION THREAD

**Discussion by axg20202** [phil-barrett/grblHAL-teensy-4.x#56] [Ideas]
Is it possible to address an RS485 module via the I2C pins and get this working with a VFD?  I quite fancy trying out RS485 spindle speed control and feedback, but preferably without the need to connect a USB device to the PC. I wonder if this approach could integrate RS485 with your board without using a USB dongle?
e.g. something like this:
https://coptonix.com/en/products/rs485-i2c-adapter-master/

**Comment by phil-barrett** [comment]
It is certainly possible but would need a grblHAL driver.  

Have you considered using a serial I/O to RS485 converter?  That could use the UART.  Several people have done that.

**Comment by axg20202** [comment]
Ahah, yes, I did not notice the uart connection. Would you mind pointing me to info on getting this configured on your board?

**Reply by phil-barrett** [reply]
The UART is labeled as Ser1 on the board and has RX, TX and Gnd shown on the board. In Teensyduino, is it serial 1, all the standard arduino (and Teensyduino) controls apply.  If you are using a HY VFD there is a driver in grblHAL that can use the UART.  It is called [plugins_spindle](https://github.com/grblHAL/Plugins_spindle/) and is a good starting point to write code if using something else. Note that Modbus uses RS485.

**Comment by axg20202** [comment]
Thanks Phil. I am using an HY VFD. I couldn't find the serial pins in the grblhal board map, unless it is being called something different. In theory should I just be able to flash with the spindle plugin and then maybe use a basic rs232 to rs485 convertor module to connect the vfd? Or will I also have to mess with pin mapping and more besides in the firmware?

**Reply by phil-barrett** [reply]
I have not worked with the spindle plugin.  Perhaps you could ask questions in the [grblHAL discussions section](https://github.com/grblHAL/core/discussions). Be sure to reference the VFD model and breakout board you are using.

**Comment by axg20202** [comment]
OK, thanks. Will do. Do you know how to assign the serial1 pins on your board to UART in grblhal? That might give me a head start if you do know.

**Reply by phil-barrett** [reply]
I have only used teensyduino (arduino) libraries and that uses Serial1.  I took a quick look at the spindle plugin and did not see where you specify the UART.  Wish I cold help you more.  Lo Siento.

**Comment by axg20202** [comment]
OK, no problem. Thanks for checking. I’ll leave you in peace! 😊
